src/com/cft/contactmerge/StreetAddressMatchLogic.java

```diff
@@ -2,13 +2,185 @@
 
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.Hashtable;
 
 public abstract class StreetAddressMatchLogic {
+    private static final ArrayList<String> poBoxTypes = new ArrayList<>(Arrays.asList("po", "box"));
+
+    private static final Hashtable<String,String> roadTypes = new Hashtable<String,String>() {{
+        put("st", "street");
+        put("street", "street");
+        put("ave", "avenue");
+        put("avenue", "avenue");
+        put("dr", "drive");
+        put("drive", "drive");
+        put("ln", "lane");
+        put("lane", "lane");
+        put("trl", "trail");
+        put("trail", "trail");
+        put("cir", "circle");
+        put("circle", "circle");
+        put("blvd", "boulevard");
+        put("boulevard", "boulevard");
+        put("rd", "road");
+        put("road", "road");
+    }};
+
+    private static final Hashtable<String,String> compassDirection = new Hashtable<String,String>() {{
```

Comment:
FYI... Some cities also use NE, NW, SE, SW. I can't say that I've ever seen them spell out North-East in an address though.

Reply:
hmmm, if you think I should include them in street address, I will. For now I'll leave them left out.